plugins/pagerduty/komand_pagerduty/actions/create_user/action.py

```diff
@@ -16,20 +16,36 @@ def __init__(self):
         )
 
     def run(self, params={}):
-        """Run action"""
-        self.logger.info("Creating user %s", params)
-        user = pypd.User.create(data=params, from_email=params["from_email"])
-
-        if user:
-            user = normalize_user(user.json)
-        else:
-            user = None
-
-        self.logger.debug("Returned: %s", user)
-
-        return {"success": not not user, "user": user or empty_user}
-
-    def test(self):
-        """Test action"""
+        """Trigger event"""
+
+        # required
+        from_email = params.get("from_email")
+        new_users_email = params.get("email")
+        name = params.get("name")
+
+        # optional
+        dict_of_optional_fields = {
+            "time_zone": params.get("time_zone", ""),
+            "color": params.get("color", ""),
+            "role": params.get("role", ""),
+            "timezone": params.get("timezone", ""),
+            "description": params.get("user_description", ""),
+            "job_title": params.get("job_title", ""),
+            "license": params.get("license", {}),
+        }
+
+        if from_email is None or new_users_email is None or name is None:
+            self.logger.warning("Please ensure a valid 'from_email', 'new_users_email' and 'name' is provided")
+            raise PluginException(
+                cause="Missing required paramaters",
+                assistance="Please ensure a valid 'from_email', 'new_users_email' and 'name' is provided",
+            )
```

Comment:
Same here, and for all other `action.py` files that are manually checking for required fields. 

By the looks of it this plugin is very old and seems to be beyond the time when we had `required` tag in spec files.

Reply:
I'll give further review once all of these are fixed up 😄

Reply:
I removed all of  these redundant error raises